R/utils.R

```diff
@@ -292,9 +292,44 @@ h_partial_fun_args <- function(fun, ..., additional_attr = list()) {
   )
 }
 
+#' Complete `character` Vector Names From Values
+#'
+#' @eval typed_param(x, character(N) | list(N),
+#'   "Value whose names should be completed from element values.")
+#'
+#' @return A named vector or list.
+#'
+#' @keywords internal
 fill_names <- function(x) {
   n <- names(x)
   is_unnamed <- if (is.null(n)) rep_len(TRUE, length(x)) else n == ""
   names(x)[is_unnamed] <- x[is_unnamed]
   x
 }
+
+#' Format 'roxygen2' Type-Annotated Parameter Definitions
```

Comment:
Hm interesting! I am not sure about this though and needs more discussion. Can we please make this a separate issue / PR? So for this one please follow the standard formatting. Thanks

Reply:
In particular, this would be one implementation way. Another would explicitly extend roxygen roclets, following https://github.com/r-lib/roxygen2/blob/main/vignettes/extending.Rmd. Plus I think it could be much better to have this done in a separate small extension package to roxygen to be able to use in all other packages we develop.

Reply:
Yeah, [I've written roclets before](https://github.com/dgkf/testex) - that's definitely the more holistic way of solving this. In this case, I'm not sure it's worth the added dependency (even if just a dev dependency), nor am I entirely sold on the value-add of embedding type annotations for parameters in a dynamic language (for example, we have plenty of functions that require a specific class 

Having a function to at least keep `mmrm`'s particular formatting consistent seemed like the right middle ground.

I can revert these changes and we can discuss whether this is worth the effort to refactor from there.

Reply:
Thanks Doug - good that you follow our conventions for now, and I suggest we discuss separately on this topic